kg_microbe/transform_utils/madin_etal/madin_etal.py

```diff
@@ -297,78 +355,148 @@ def run(self, data_file: Union[Optional[Path], Optional[str]] = None, show_statu
+                                    == chebi_result_for_tax_id[SUBJECT_LABEL_COLUMN]
+                                )
+                                exact_match_chebi_df = chebi_result_for_tax_id[
+                                    exact_condition_chebi
+                                ]
+                                if not exact_match_chebi_df.empty:
+                                    chebi_result_for_tax_id = exact_match_chebi_df
+                                for row in chebi_result_for_tax_id.iterrows():
+                                    carbon_substrate_nodes.append(
+                                        [
+                                            row[1].object_id,
+                                            CARBON_SUBSTRATE_CATEGORY,
+                                            row[1].object_label,
+                                        ]
+                                    )
+                                    tax_carbon_substrate_edge.append(
+                                        [
+                                            tax_id,
+                                            NCBI_TO_CARBON_SUBSTRATE_EDGE,
+                                            row[1].object_id,
+                                            BIOLOGICAL_PROCESS,
+                                        ]
+                                    )
 
                         node_writer.writerows(carbon_substrate_nodes)
                         edge_writer.writerows(tax_carbon_substrate_edge)
 
+                    # block handling "cell shape" column from Madin etal dataset/CSV sheet
                     cell_shape = (
                         None
                         if filtered_row[CELL_SHAPE_COLUMN] == "NA"
                         else filtered_row[CELL_SHAPE_COLUMN]
                     )
                     if cell_shape:
-                        cell_shape_node = [
-                            SHAPE_PREFIX + cell_shape,
-                            PHENOTYPIC_CATEGORY,
-                            cell_shape,
-                        ]
-                        tax_to_cell_shape_edge = [
-                            tax_id,
-                            NCBI_TO_SHAPE_EDGE,
-                            SHAPE_PREFIX + cell_shape,
-                            HAS_PHENOTYPE,
-                        ]
+                        # First try to find mapping in METPO
+                        metpo_mapping = self.madin_metpo_mappings.get(cell_shape.strip(), None)
+                        # print(f"Cell shape: {cell_shape}, METPO mapping: {metpo_mapping}")
+                        if metpo_mapping:
+                            # create cell shape node and edge to tax_id
+                            # use `biolink equivalent` for 'category' and
+                            # for 'predicate', `biolink equivalent` from properties
+                            category = metpo_mapping.get("biolink_equivalent", PHENOTYPIC_CATEGORY)
```

Comment:
@sujaypatil96 another node category case PHENOTYPIC_CATEGORY

let's discuss and maybe this is one more issue after this PR